### ISSUE TITLE
Can't build on Cygwin

### ISSUE THREAD

**Issue by giann** [bitlbee/bitlbee-facebook#62]
Can't seem to be able to build it on cygwin:

```
Making all in facebook
make[1]: Entering directory '/home/bgiannan/bitlbee-facebook/facebook'
  CCLD     facebook.la
libtool:   error: can't build i686-pc-cygwin shared library unless -no-undefined is specified
Makefile:434: recipe for target 'facebook.la' failed
make[1]: *** [facebook.la] Error 1
make[1]: Leaving directory '/home/bgiannan/bitlbee-facebook/facebook'
Makefile:386: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

I tried feeding -no-undefined to libtool but it fails when linking.

**Comment by jgeboski** [comment]
@EionRobb I know bitlbee is out of your realm, but would you have any ideas on this?

**Comment by EionRobb** [comment]
Looks like you need to add `-no-undefined` if you're building with cygwin ;)
Isn't it possible to add CFLAGS when you `make`, like `CFLAGS += -no-undefined make` or something?

**Comment by jgeboski** [comment]
Might be able to pass it via `LDFLAGS`.

**Comment by dequis** [comment]
From irc, after passing `LDFLAGS=-no-undefined ./autogen.sh`:

```
$ make
Making all in facebook
make[1]: se entra en el directorio '/tmp/bitlbee-facebook/facebook'
  CCLD     facebook.la
.libs/facebook_la-facebook.o: In function `fb_cmd_account':
/tmp/bitlbee-facebook/facebook/facebook.c:906: undefined reference to `account_get'
/tmp/bitlbee-facebook/facebook/facebook.c:925: undefined reference to `account_get'
/tmp/bitlbee-facebook/facebook/facebook.c:892: undefined reference to `irc_rootmsg'
/tmp/bitlbee-facebook/facebook/facebook.c:901: undefined reference to `irc_rootmsg'
/tmp/bitlbee-facebook/facebook/facebook.c:935: undefined reference to `irc_rootmsg'
/tmp/bitlbee-facebook/facebook/facebook.c:920: undefined reference to `irc_rootmsg'
/tmp/bitlbee-facebook/facebook/facebook.c:915: undefined reference to `irc_rootmsg'
.libs/facebook_la-facebook.o:/tmp/bitlbee-facebook/facebook/facebook.c:910: more undefined references to `irc_rootmsg' follow
.libs/facebook_la-facebook.o: In function `fb_groupchat_new':
/tmp/bitlbee-facebook/facebook/facebook.c:43: undefined reference to `bee_chat_by_title'
/tmp/bitlbee-facebook/facebook/facebook.c:54: undefined reference to `imcb_chat_new'
/tmp/bitlbee-facebook/facebook/facebook.c:55: undefined reference to `imcb_chat_name_hint'
/tmp/bitlbee-facebook/facebook/facebook.c:61: undefined reference to `set_setstr'
/tmp/bitlbee-facebook/facebook/facebook.c:62: undefined reference to `set_setstr'
/tmp/bitlbee-facebook/facebook/facebook.c:63: undefined reference to `set_setstr'
/tmp/bitlbee-facebook/facebook/facebook.c:64: undefined reference to `set_setstr'
/tmp/bitlbee-facebook/facebook/facebook.c:67: undefined reference to `imcb_chat_free'
/tmp/bitlbee-facebook/facebook/facebook.c:70: undefined reference to `imcb_chat_new'
/tmp/bitlbee-facebook/facebook/facebook.c:77: undefined reference to `imcb_chat_add_buddy'
/tmp/bitlbee-facebook/facebook/facebook.c:44: undefined reference to `imcb_error'
.libs/facebook_la-facebook.o: In function `fb_cmd_fbcreate':
/tmp/bitlbee-facebook/facebook/facebook.c:986: undefined reference to `irc_user_by_name'
/tmp/bitlbee-facebook/facebook/facebook.c:996: undefined reference to `irc_rootmsg'
.libs/facebook_la-facebook.o: In function `fb_login':
/tmp/bitlbee-facebook/facebook/facebook.c:715: undefined reference to `imcb_log'
/tmp/bitlbee-facebook/facebook/facebook.c:710: undefined reference to `imcb_log'
.libs/facebook_la-facebook.o: In function `fb_cb_api_auth':
/tmp/bitlbee-facebook/facebook/facebook.c:134: undefined reference to `imcb_log'
.libs/facebook_la-facebook.o: In function `fb_cb_api_typing':
/tmp/bitlbee-facebook/facebook/facebook.c:606: undefined reference to `imcb_buddy_typing'
.libs/facebook_la-facebook.o: In function `fb_cb_api_threads':
/tmp/bitlbee-facebook/facebook/facebook.c:560: undefined reference to `imcb_log'
/tmp/bitlbee-facebook/facebook/facebook.c:589: undefined reference to `imcb_log'
.libs/facebook_la-facebook.o: In function `fb_cb_api_thread_create':
/tmp/bitlbee-facebook/facebook/facebook.c:534: undefined reference to `imcb_log'
.libs/facebook_la-facebook.o: In function `fb_cb_api_thread':
/tmp/bitlbee-facebook/facebook/facebook.c:477: undefined reference to `bee_chat_by_title'
/tmp/bitlbee-facebook/facebook/facebook.c:499: undefined reference to `imcb_chat_topic'
/tmp/bitlbee-facebook/facebook/facebook.c:511: undefined reference to `bee_user_by_handle'
/tmp/bitlbee-facebook/facebook/facebook.c:519: undefined reference to `imcb_chat_add_buddy'
/tmp/bitlbee-facebook/facebook/facebook.c:496: undefined reference to `imcb_chat_topic'
/tmp/bitlbee-facebook/facebook/facebook.c:514: undefined reference to `bee_user_new'
/tmp/bitlbee-facebook/facebook/facebook.c:515: undefined reference to `imcb_buddy_nick_hint'
/tmp/bitlbee-facebook/facebook/facebook.c:516: undefined reference to `imcb_rename_buddy'
.libs/facebook_la-facebook.o: In function `fb_cb_api_presences':
/tmp/bitlbee-facebook/facebook/facebook.c:447: undefined reference to `bee_user_by_handle'
/tmp/bitlbee-facebook/facebook/facebook.c:458: undefined reference to `imcb_buddy_status'
.libs/facebook_la-facebook.o: In function `fb_eval_open':
/tmp/bitlbee-facebook/facebook/facebook.c:612: undefined reference to `is_bool'
/tmp/bitlbee-facebook/facebook/facebook.c:613: undefined reference to `SET_INVALID'
.libs/facebook_la-facebook.o: In function `fb_cb_api_events':
/tmp/bitlbee-facebook/facebook/facebook.c:336: undefined reference to `imcb_chat_remove_buddy'
/tmp/bitlbee-facebook/facebook/facebook.c:313: undefined reference to `bee_chat_by_title'
/tmp/bitlbee-facebook/facebook/facebook.c:327: undefined reference to `bee_user_by_handle'
/tmp/bitlbee-facebook/facebook/facebook.c:332: undefined reference to `imcb_chat_add_buddy'
/tmp/bitlbee-facebook/facebook/facebook.c:323: undefined reference to `imcb_chat_topic'
.libs/facebook_la-facebook.o: In function `fb_cb_api_error':
/tmp/bitlbee-facebook/facebook/facebook.c:289: undefined reference to `imcb_error'
.libs/facebook_la-facebook.o: In function `fb_cb_api_contacts':
/tmp/bitlbee-facebook/facebook/facebook.c:223: undefined reference to `imcb_add_buddy'
/tmp/bitlbee-facebook/facebook/facebook.c:224: undefined reference to `imcb_buddy_nick_hint'
/tmp/bitlbee-facebook/facebook/facebook.c:225: undefined reference to `imcb_rename_buddy'
/tmp/bitlbee-facebook/facebook/facebook.c:227: undefined reference to `imcb_buddy_by_handle'
/tmp/bitlbee-facebook/facebook/facebook.c:258: undefined reference to `set_getint'
/tmp/bitlbee-facebook/facebook/facebook.c:261: undefined reference to `set_setint'
/tmp/bitlbee-facebook/facebook/facebook.c:253: undefined reference to `imcb_log'
/tmp/bitlbee-facebook/facebook/facebook.c:248: undefined reference to `imcb_remove_buddy'
.libs/facebook_la-facebook.o: In function `fb_cb_api_connect':
/tmp/bitlbee-facebook/facebook/facebook.c:150: undefined reference to `imcb_connected'
/tmp/bitlbee-facebook/facebook/facebook.c:152: undefined reference to `set_getbool'
.libs/facebook_la-facebook.o: In function `fb_init':
/tmp/bitlbee-facebook/facebook/facebook.c:624: undefined reference to `set_add'
/tmp/bitlbee-facebook/facebook/facebook.c:627: undefined reference to `set_add'
/tmp/bitlbee-facebook/facebook/facebook.c:630: undefined reference to `set_add'
/tmp/bitlbee-facebook/facebook/facebook.c:633: undefined reference to `set_add'
/tmp/bitlbee-facebook/facebook/facebook.c:636: undefined reference to `set_add'
.libs/facebook_la-facebook.o:/tmp/bitlbee-facebook/facebook/facebook.c:639: more undefined references to `set_add' follow
.libs/facebook_la-facebook.o: In function `fb_init':
/tmp/bitlbee-facebook/facebook/facebook.c:643: undefined reference to `set_eval_bool'
/tmp/bitlbee-facebook/facebook/facebook.c:643: undefined reference to `set_add'
/tmp/bitlbee-facebook/facebook/facebook.c:644: undefined reference to `set_eval_bool'
/tmp/bitlbee-facebook/facebook/facebook.c:644: undefined reference to `set_add'
/tmp/bitlbee-facebook/facebook/facebook.c:645: undefined reference to `set_eval_int'
/tmp/bitlbee-facebook/facebook/facebook.c:645: undefined reference to `set_add'
.libs/facebook_la-facebook.o: In function `fb_channel_join':
/tmp/bitlbee-facebook/facebook/facebook.c:94: undefined reference to `set_getstr'
/tmp/bitlbee-facebook/facebook/facebook.c:100: undefined reference to `set_getstr'
.libs/facebook_la-facebook.o: In function `fb_cmd_fbjoin':
/tmp/bitlbee-facebook/facebook/facebook.c:1043: undefined reference to `irc_rootmsg'
/tmp/bitlbee-facebook/facebook/facebook.c:1033: undefined reference to `irc_rootmsg'
.libs/facebook_la-facebook.o: In function `fb_cb_api_messages':
/tmp/bitlbee-facebook/facebook/facebook.c:370: undefined reference to `set_getbool'
/tmp/bitlbee-facebook/facebook/facebook.c:371: undefined reference to `set_find'
/tmp/bitlbee-facebook/facebook/facebook.c:372: undefined reference to `set_getstr'
/tmp/bitlbee-facebook/facebook/facebook.c:374: undefined reference to `is_bool'
/tmp/bitlbee-facebook/facebook/facebook.c:374: undefined reference to `bool2int'
/tmp/bitlbee-facebook/facebook/facebook.c:394: undefined reference to `bee_user_by_handle'
/tmp/bitlbee-facebook/facebook/facebook.c:406: undefined reference to `imcb_buddy_msg'
/tmp/bitlbee-facebook/facebook/facebook.c:411: undefined reference to `bee_chat_by_title'
/tmp/bitlbee-facebook/facebook/facebook.c:426: undefined reference to `imcb_chat_msg'
/tmp/bitlbee-facebook/facebook/facebook.c:415: undefined reference to `bee_chat_by_title'
.libs/facebook_la-facebook.o: In function `fb_cb_api_contact':
/tmp/bitlbee-facebook/facebook/facebook.c:168: undefined reference to `bee_user_by_handle'
/tmp/bitlbee-facebook/facebook/facebook.c:169: undefined reference to `bee_user_new'
/tmp/bitlbee-facebook/facebook/facebook.c:170: undefined reference to `imcb_buddy_nick_hint'
/tmp/bitlbee-facebook/facebook/facebook.c:171: undefined reference to `imcb_rename_buddy'
.libs/facebook_la-facebook.o: In function `init_plugin':
/tmp/bitlbee-facebook/facebook/facebook.c:1079: undefined reference to `register_protocol'
/tmp/bitlbee-facebook/facebook/facebook.c:1081: undefined reference to `root_command_add'
/tmp/bitlbee-facebook/facebook/facebook.c:1082: undefined reference to `root_command_add'
/tmp/bitlbee-facebook/facebook/facebook.c:1083: undefined reference to `root_command_add'
.libs/facebook_la-facebook.o: In function `fb_chat_leave':
/tmp/bitlbee-facebook/facebook/facebook.c:824: undefined reference to `imcb_chat_free'
.libs/facebook_la-facebook.o: In function `fb_cb_api_threads':
/tmp/bitlbee-facebook/facebook/facebook.c:555: undefined reference to `imcb_log'
.libs/facebook_la-facebook.o: In function `fb_cb_api_thread_create':
/tmp/bitlbee-facebook/facebook/facebook.c:535: undefined reference to `imcb_log'
.libs/facebook_la-facebook.o: In function `fb_cb_api_error':
/tmp/bitlbee-facebook/facebook/facebook.c:290: undefined reference to `imc_logout'
.libs/facebook_la-facebook-api.o: In function `fb_api_cb_http_bool':
/tmp/bitlbee-facebook/facebook/facebook-api.c:784: undefined reference to `bool2int'
.libs/facebook_la-facebook-data.o: In function `fb_data_dispose':
/tmp/bitlbee-facebook/facebook/facebook-data.c:54: undefined reference to `b_event_remove'
/tmp/bitlbee-facebook/facebook/facebook-data.c:61: undefined reference to `imcb_chat_free'
.libs/facebook_la-facebook-data.o: In function `fb_data_new':
/tmp/bitlbee-facebook/facebook/facebook-data.c:104: undefined reference to `imcb_new'
.libs/facebook_la-facebook-data.o: In function `fb_data_load':
/tmp/bitlbee-facebook/facebook/facebook-data.c:126: undefined reference to `set_getstr'
/tmp/bitlbee-facebook/facebook/facebook-data.c:139: undefined reference to `set_getstr'
/tmp/bitlbee-facebook/facebook/facebook-data.c:151: undefined reference to `set_getstr'
.libs/facebook_la-facebook-data.o: In function `fb_data_save':
/tmp/bitlbee-facebook/facebook/facebook-data.c:187: undefined reference to `set_setstr'
/tmp/bitlbee-facebook/facebook/facebook-data.c:197: undefined reference to `set_setstr'
/tmp/bitlbee-facebook/facebook/facebook-data.c:206: undefined reference to `set_setstr'
.libs/facebook_la-facebook-data.o: In function `fb_data_clear_timeout':
/tmp/bitlbee-facebook/facebook/facebook-data.c:323: undefined reference to `b_event_remove'
.libs/facebook_la-facebook-data.o: In function `fb_data_add_timeout':
/tmp/bitlbee-facebook/facebook/facebook-data.c:305: undefined reference to `b_timeout_add'
.libs/facebook_la-facebook-http.o: In function `fb_http_request_dispose':
/tmp/bitlbee-facebook/facebook/facebook-http.c:101: undefined reference to `http_close'
.libs/facebook_la-facebook-http.o: In function `fb_http_urlcmp':
/tmp/bitlbee-facebook/facebook/facebook-http.c:531: undefined reference to `url_set'
/tmp/bitlbee-facebook/facebook/facebook-http.c:531: undefined reference to `url_set'
.libs/facebook_la-facebook-http.o: In function `fb_http_request_new':
/tmp/bitlbee-facebook/facebook/facebook-http.c:261: undefined reference to `url_set'
.libs/facebook_la-facebook-http.o: In function `fb_http_request_send':
/tmp/bitlbee-facebook/facebook/facebook-http.c:497: undefined reference to `http_dorequest'
.libs/facebook_la-facebook-http.o: In function `fb_http_values_get_bool':
/tmp/bitlbee-facebook/facebook/facebook-http.c:756: undefined reference to `bool2int'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_timeout_clear':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:285: undefined reference to `b_event_remove'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_ping':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:322: undefined reference to `b_timeout_add'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_timeout_clear':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:285: undefined reference to `b_event_remove'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_timeout':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:296: undefined reference to `b_timeout_add'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_close':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:224: undefined reference to `b_event_remove'
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:229: undefined reference to `b_event_remove'
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:234: undefined reference to `b_event_remove'
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:239: undefined reference to `ssl_disconnect'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_cb_write':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:508: undefined reference to `ssl_write'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_cb_open':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:572: undefined reference to `ssl_getfd'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_timeout_clear':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:285: undefined reference to `b_event_remove'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_cb_open':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:574: undefined reference to `b_input_add'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_open':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:588: undefined reference to `ssl_connect'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_write':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:551: undefined reference to `ssl_getfd'
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:555: undefined reference to `b_input_add'
.libs/facebook_la-facebook-mqtt.o: In function `fb_mqtt_cb_read':
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:370: undefined reference to `ssl_read'
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:342: undefined reference to `ssl_read'
/tmp/bitlbee-facebook/facebook/facebook-mqtt.c:354: undefined reference to `ssl_read'
.libs/facebook_la-facebook-util.o: In function `fb_util_rand_uuid':
/tmp/bitlbee-facebook/facebook/facebook-util.c:267: undefined reference to `sha1_init'
/tmp/bitlbee-facebook/facebook/facebook-util.c:268: undefined reference to `random_bytes'
/tmp/bitlbee-facebook/facebook/facebook-util.c:269: undefined reference to `sha1_append'
/tmp/bitlbee-facebook/facebook/facebook-util.c:270: undefined reference to `sha1_random_uuid'
collect2: error: ld returned 1 exit status
Makefile:434: fallo en las instrucciones para el objetivo 'facebook.la'
make[1]: *** [facebook.la] Error 1
make[1]: se sale del directorio '/tmp/bitlbee-facebook/facebook'
Makefile:386: fallo en las instrucciones para el objetivo 'all-recursive'
make: *** [all-recursive] Error 1
```

**Comment by giann** [comment]
I managed to built it. But I get 'Unknown protocol' when adding an account.

**Comment by dequis** [comment]
What did you do to build it?

**Comment by giann** [comment]
I modified the libtool script provided by this repo and forced `allow_undefined`. `-no-undefined` wasn't recognised for some reason.

**Comment by dequis** [comment]
The main issue here was that the plugin didn't have anything to link to, that's fixed with this bitlbee commit: https://github.com/bitlbee/bitlbee/commit/54b2a367f6986a74fd6b4e50a0ae2c5cdb3a40cb (only learned about this recently thanks to someone in #cygwin)

The other, less exciting issue (no-undefined), is fixed by 310de485e9597df35e94cd4020f8b23121cb3904 in this repo

So just building the latest git versions of both bitlbee and this plugin should be enough to get it working. When building this new version of bitlbee it's important to repeat the whole process from `./configure` to `make install-dev` as there were important changes to bitlbee.pc
